elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/dynamic/compile/ElideDynamicEntityCompiler.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.spring.dynamic.compile;
+
+import com.yahoo.elide.contrib.dynamicconfighelpers.model.ElideSecurityConfig;
+import com.yahoo.elide.contrib.dynamicconfighelpers.model.ElideTableConfig;
+import com.yahoo.elide.contrib.dynamicconfighelpers.parser.ElideConfigParser;
+import com.yahoo.elide.contrib.dynamicconfighelpers.parser.handlebars.HandlebarsHydrator;
+import com.google.common.collect.Sets;
+
+import org.mdkt.compiler.InMemoryJavaCompiler;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Compiles dynamic model pojos generated from hjson files.
+ *
+ */
+@Slf4j
+public class ElideDynamicEntityCompiler {
+
+    public static ArrayList<String> classNames = new ArrayList<String>();
+
+    public static final String PACKAGE_NAME = "com.yahoo.elide.contrib.dynamicconfig.model.";
+    private Map<String, Class<?>> compiledObjects;
+
+    private InMemoryJavaCompiler compiler = InMemoryJavaCompiler.newInstance();
+
+    private Map<String, String> tableClasses = new HashMap<String, String>();
+    private Map<String, String> securityClasses = new HashMap<String, String>();
+
+    /**
+     * Parse dynamic config path.
+     * @param path : Dynamic config hjsons root location
+     */
+    public ElideDynamicEntityCompiler(String path) {
+
+        ElideTableConfig tableConfig = new ElideTableConfig();
+        ElideSecurityConfig securityConfig = new ElideSecurityConfig();
+        ElideConfigParser elideConfigParser = new ElideConfigParser();
+        HandlebarsHydrator hydrator = new HandlebarsHydrator();
+
+        try {
+
+            elideConfigParser.parseConfigPath(path);
```

Comment:
What handles the variable substitution for the variables.hjson?

Reply:
Variable substitution happens in  elide contrib dynamic config helpers https://github.com/yahoo/elide/blob/e36cc01d6976ce0f511e47babd3b5b2e313b2138/elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/DynamicConfigHelpersUtil.java#L176